services/slam/builtin/builtin.go

```diff
@@ -457,6 +479,44 @@ func (slamSvc *builtIn) GetMap(
 	if err != nil {
 		return "", nil, nil, err
 	}
+	reqPCMap := &pb.GetPointCloudMapRequest{
+		Name: name,
+	}
+
+	var imData image.Image
+	var vObj *vision.Object
+
+	// TODO: Once RSDK-1053 (https://viam.atlassian.net/browse/RSDK-1066) is complete the original code that extracts
+	// the map will be removed and GetMap will be changed to GetPointCloudMap
+	if slamSvc.dev {
+		slamSvc.logger.Debug("IN DEV MODE (map request)")
+
+		if mimeType != rdkutils.MimeTypePCD {
+			return "", nil, nil, errors.New("non-pcd return type is impossible in while in dev mode")
+		}
+
+		resp, err := slamSvc.clientAlgo.GetPointCloudMap(ctx, reqPCMap)
+
+		if err != nil {
+			return "", imData, vObj, errors.Errorf("error getting SLAM map (%v) : %v", mimeType, err)
+		}
+		pointcloudData := resp.GetPointCloudPcd()
+		if pointcloudData == nil {
+			return "", nil, nil, errors.New("get map read pointcloud unavailable")
+		}
+		pc, err := pc.ReadPCD(bytes.NewReader(pointcloudData))
+		if err != nil {
+			return "", nil, nil, errors.Wrap(err, "get map read pointcloud failed")
+		}
+
+		vObj, err = vision.NewObject(pc)
+		if err != nil {
+			return "", nil, nil, errors.Wrap(err, "get map creating vision object failed")
+		}
+
+		return rdkutils.MimeTypePCD, imData, vObj, nil
+	}
```

Comment:
[nit] have an `} else {` statement here and put everything below this into the `else` scope. This will make your use of the `dev` flag in `GetMap` consistent with how you're using it in `Position`.